pf4j/src/main/java/org/pf4j/processor/ExtensionAnnotationProcessor.java

```diff
@@ -85,53 +85,61 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
         }
 
         info("Processing @%s", Extension.class);
-		for (Element element : roundEnv.getElementsAnnotatedWith(Extension.class)) {
-            // check if @Extension is put on class and not on method or constructor
-            if (!(element instanceof TypeElement)) {
-				continue;
-			}
-
-            // check if class extends/implements an extension point
-            if (!isExtension(element.asType())) {
+        readExtensionsFromAnnotatedClasses(roundEnv);
+
+        // read old extensions
+        oldExtensions = storage.read();
+        copyDataFromTo(oldExtensions, extensions);
+
+        // write extensions
+        storage.write(extensions);
+
+		return false;
+	}
+
+    private void readExtensionsFromAnnotatedClasses(RoundEnvironment roundEnv) {
+        for (Element element : roundEnv.getElementsAnnotatedWith(Extension.class)) {
+// check if @Extension is put on class and not on method or constructor
```

Comment:
it's strange that the code is not indented. I cannot follow/tack the modifications